modules/config/src/main/scala/scala/cli/config/Key.scala

```diff
@@ -83,32 +84,35 @@ object Key {
         cause
       )
 
-  final class MalformedEntry(
+  private final class MalformedEntry(
     entry: Key[_],
     messages: ::[String]
   ) extends EntryError(
         s"Malformed entry ${entry.fullName}, " +
           messages.mkString(", ")
       )
 
-  private val stringCodec: JsonValueCodec[String]   = JsonCodecMaker.make
-  private val booleanCodec: JsonValueCodec[Boolean] = JsonCodecMaker.make
+  abstract class KeyWithJsonCodec[T] extends Key[T] {
+    def jsonCodec: JsonValueCodec[T]
```

Comment:
I would prefer to avoid returning `JsonValueCodec[T]` from a function, since it can generate the same codec multiple times for only one type.

It might be better to initialize this codec in the companion object and define it as `implicit lazy val` to ensure that it is only initialized once.

Reply:
should be fixed